logstash-core/lib/logstash/instrument/collector.rb

```diff
@@ -61,6 +61,9 @@ def push(namespaces_path, key, type, *metric_type_params)
       end
     end
 
+    def clear
```

Comment:
Yes, I wanted to make sure each run were isolated, since the collector is still singleton. I have added this method.